# Individual Planning Stage

https://github.com/YuanYuan-06/dsci-Individualplan.git

## Loading package

In [1]:
library(tidyverse)
library(dplyr)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


## Reading data

### reading "players" data

In [2]:
url_player = "http://drive.google.com/uc?export=download&id=1EHNNhxB-0P13COf3nGGV7hi2Y96ay5hq"
players <- read_csv(url_player)
head(players)
dim(players)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

[1] 196   7

### reading "sessions" data

In [3]:
url_sessions = "http://drive.google.com/uc?export=download&id=19U6P0gnqzlpL4NFfIgrG9riMbup52klE"
sessions <- read_csv(url_sessions)
head(sessions)
dim(sessions)

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.72e+12            1.72e+12         
2 17/06/2024 23:33 17/06/2024 23:46 1.72e+12            1.72e+12         
3 25/07/2024 17:34 25/07/2024 17:57 1.72e+12            1.72e+12         
4 25/07/2024 03:22 25/07/2024 03:58 1.72e+12            1.72e+12         
5 25/05/2024 16:01 25/05/2024 16:12 1.72e+12            1.72e+12         
6 23/06/2024 15:08 23/06/2024 17:10 1.72e+12            1.72e+12

[1] 1535    5

## Data organization

### organizing"players" data

In [4]:
distinct_experience <- distinct(players,experience)
distinct_subscribe <- distinct(players,subscribe)
distinct_gender<- distinct(players,gender)

distinct_experience
distinct_subscribe
distinct_gender

experience
1 Pro       
2 Veteran   
3 Amateur   
4 Regular   
5 Beginner

subscribe
1  TRUE    
2 FALSE

gender           
1 Male             
2 Female           
3 Non-binary       
4 Prefer not to say
5 Agender          
6 Two-Spirited     
7 Other

In [5]:
player_max<- players |>
summarize(across(c(played_hours,Age),~max(.x, na.rm=TRUE)),statistic = "max")

player_min<- players |>
summarize(across(c(played_hours,Age),~min(.x, na.rm=TRUE)),statistic = "min")

player_mean<- players |>
summarize(across(c(played_hours,Age),~mean(.x, na.rm=TRUE)),statistic = "mean")

player_max
player_min
player_mean

played_hours Age statistic
1 223.1        58  max

played_hours Age statistic
1 0            9   min

played_hours Age      statistic
1 5.845918     21.13918 mean

In [6]:
players_missing <- players|>
summarize(across(experience:Age, ~sum(is.na(.x))))
players_missing

experience subscribe hashedEmail played_hours name gender Age
1 0          0         0           0            0    0      2

In [7]:
experience_frequency <- players|>
count(experience)|>
mutate(percent = n/sum(n) *100)|>
arrange(desc(n))

gender_frequency <- players|>
count(gender)|>
mutate(percent = n/sum(n) *100)|>
arrange(desc(n))

subscribe_frequency <- players|>
count(subscribe)|>
mutate(percent = n/sum(n) *100)|>
arrange(desc(n))

experience_frequency
gender_frequency
subscribe_frequency

experience n  percent  
1 Amateur    63 32.142857
2 Veteran    48 24.489796
3 Regular    36 18.367347
4 Beginner   35 17.857143
5 Pro        14  7.142857

gender            n   percent   
1 Male              124 63.2653061
2 Female             37 18.8775510
3 Non-binary         15  7.6530612
4 Prefer not to say  11  5.6122449
5 Two-Spirited        6  3.0612245
6 Agender             2  1.0204082
7 Other               1  0.5102041

subscribe n   percent 
1  TRUE     144 73.46939
2 FALSE      52 26.53061

### organizing "sessions" data

In [8]:
distinct_experience <- distinct(players,experience)
distinct_subscribe <- distinct(players,subscribe)
distinct_gender<- distinct(players,gender)

distinct_experience
distinct_subscribe
distinct_gender

experience
1 Pro       
2 Veteran   
3 Amateur   
4 Regular   
5 Beginner

subscribe
1  TRUE    
2 FALSE

gender           
1 Male             
2 Female           
3 Non-binary       
4 Prefer not to say
5 Agender          
6 Two-Spirited     
7 Other

## Data description

### describing "players" data

The data have 196 observations and 9 variables.  
The variables are:
1. `experience`(character): User's game experience level, consisting of `Pro`, `Veteran`, `Amateur`, `Regular`, and `Beginner`, with `Amateur` (32.14%) and `Pro` (7.14%) as the most and least frequent. 
2. `subscribe`(logical): Indicate whether the user has subscribed to the service, consisting of `TRUE` and `FALSE`. 73.47% players subscribed to game information, 26.53% did not.  
3. `hashedEmail`(character): Hashing encoded email, which is the encrypted form of the user's email address.
4. `played_hours`(double): User's online gaming time in hours, ranging from `0` to `223.1` hours, with a mean of 5.85 hours. 
5. `name`(character): Recorded user's name
6. `gender`(character): Recorded user's gender, consisting of `Male`, `Female`, `Non-binary`, `Prefer not to say`, `Agender`, `Two-Spirited`, and `Other`, with `Male` (63.27%) and `Other` (0.51%) as the most and least frequent. 
7. `Age`(double): Recorded user's age, ranging from `9` to `58`, with a mean of 21.14. 